src/shop/ordering/orders.js

```diff
@@ -396,6 +545,278 @@ bloombox.shop.order.Order.prototype.addItem = function(item) {
+ * @package
+ */
+bloombox.shop.order.Order.fromProto = function(protob) {
+  // @TODO: decode additional properties here once they are available
+  let targetId = protob.getId();
+  let underlyingOrderType = protob.getType();
+  let underlyingStatus = protob.getStatus();
+
+  let status = bloombox.shop.order.Order.inflateStatus(underlyingStatus);
+  let type = bloombox.shop.order.Order.inflateType(underlyingOrderType);
+  let order = new bloombox.shop.order.Order(type, null, null);
+  order.setId(targetId);
+  order.setStatus(status);
+
+  return order;
+};
+
+
+/**
+ * Retrieve a copy of this order from the server, and update the properties on
+ * it accordingly.
+ *
+ * @param {bloombox.shop.OrderGetCallback} callback Callback to dispatch once
+ *        the order update has completed.
+ * @throws {bloombox.shop.order.OrderException} If no order ID is specified on
+ *        the current order, indicating it has not successfully been submitted
+ *        to the shop service yet.
+ * @export
+ */
+bloombox.shop.order.Order.prototype.update = function(callback) {
+  // check for a valid order ID
+  if (this.id === null)
+    throw new bloombox.shop.order.OrderException(
+      'Cannot update order with no ID.');
+
+  let subject = this;
+
+  bloombox.shop.order.Order.retrieve(this.id, function(err, order) {
+    if (err !== null || order === null) {
+      // could not retrieve order update
+      if (err !== null) {
+        // an error occurred
+        bloombox.logging.error(
+          'Failed to retrieve updated order from server. Got error: "' +
+          err.toString() + ', for order ID "' + subject.id + '".', {
+            'error': err,
+            'order': order
+          });
+        callback(err, null);
+      } else {
+        // got no error and no order
+        bloombox.logging.error(
+          'Failed to retrieve updated order from server. Additionally ' +
+          'got no error. Order ID "' + subject.id + '".', {
+            'error': err,
+            'order': order,
+            'subject': subject
+          });
+        callback(null, null);
+      }
+    } else {
+      // update the order from the response
+      if (order.getStatus())
+        subject.setStatus(
+          bloombox.shop.order.Order.inflateStatus(order.getStatus()));
+
+      // dispatch callback, we're done
+      callback(null, subject);
+    }
+  });
+};
+
+
+/**
+ * Retrieve an order by its key, which is returned after submission of an order
+ * to the shop service.
+ *
+ * @param {string} key Key for the desired order.
+ * @param {bloombox.shop.OrderGetCallback} callback Callback to dispatch with
+ *        the resulting order.
+ * @throws {bloombox.shop.order.OrderException} If no order ID is specified on
+ *        the current order, indicating it has not successfully been submitted
+ *        to the shop service yet.
+ * @export
+ */
+bloombox.shop.order.Order.retrieve = function(key, callback) {
```

Comment:
you can also retrieve an order with:
`Order.retrieve("abc123", function(error, order) { ... });`

beware, for now orders retrieved this way only expose their ID, their type (`DELIVERY`/`PICKUP`) and their status (enumerated above in the new `bloombox.shop.order.Status`).